app/src/main/java/com/example/benchproject/data/movie/details/repo/MovieDetailsDataRepository.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.benchproject.data.movie.details.repo
+
+import com.example.benchproject.data.movies.common.MoviesApi
+import com.example.benchproject.domain.movie.details.entity.GenresModel
+import com.example.benchproject.domain.movie.details.entity.MovieDetailsModel
+import com.example.benchproject.domain.movie.details.repo.MovieDetailsRepository
+import javax.inject.Inject
+
+class MovieDetailsDataRepository @Inject constructor(
+    private val apiService: MoviesApi
+) : MovieDetailsRepository {
+    override suspend fun getMovieDetails(movieId: Int): Result<MovieDetailsModel> =
+        runCatching {
+            val movieDetailsRemote = apiService.getMovieDetails(movieId)
+            MovieDetailsModel(
+                movieDetailsRemote.id,
+                BASE_URL_FOR_IMAGE + movieDetailsRemote.imageBackgroundPath,
+                movieDetailsRemote.name,
+                movieDetailsRemote.genres.map { genresRemote ->
+                    GenresModel(genresRemote.id, genresRemote.name)
+                },
+                movieDetailsRemote.description,
+                movieDetailsRemote.budget,
+                movieDetailsRemote.rating,
+                movieDetailsRemote.releaseDate,
+                BASE_URL_FOR_IMAGE + movieDetailsRemote.imagePath
+            )
+        }.onFailure {
+            it.message
+        }
+
+    companion object {
```

Comment:
It should probably be private

Reply:
Or as I remember correctly it was placed also in another repo, so you should extract it to some common place if it's used in a few places. Don't repeat yourself.

Reply:
Ok I added to movies.common like moviesConstans